### PR TITLE
 ex16_ch4.py

### PR DESCRIPTION
**Pull request by amymhaddad**
Instructions: Write a program that asks the user for their age and compare it to the legal driving age of sixteen. If the user is sixteen or older, then the program should display “You are old enough to legally drive.” If the user is under sixteen, the program should display “You are not old enough to legally drive."

Constraints
• Use a single output statement.
• If the user enters a number that’s less than zero or enters non-numeric data, display an error message that asks the user to enter a valid age.

### CODE REVIEW

File: ex16_ch4.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Thu May 10 15:34:22 2018
+
+@author: amyhaddad
+"""
+
+#step1: state constant
+LEGAL_DRIVING_AGE = 16
```
Comment (paulghaddad) [inline ex16_ch4.py]: Great use of a constant and a great name!

File: ex16_ch4.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+#step2: get input from user on their age
+user_age = int(input("Enter your age: "))
+
+#step3: set up first conditional
+if user_age >= LEGAL_DRIVING_AGE:
+    print("You're old enough to drive legally.")
+
+#step4: set up second conditional if user enters a number less than 0
+if user_age < 0:
```
Comment (paulghaddad) [inline ex16_ch4.py]: I'd personally test this before checking the age against the legal driving age. If it isn't a valid age, there's no use in proceeding further.

File: ex16_ch4.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+#step3: set up first conditional
+if user_age >= LEGAL_DRIVING_AGE:
+    print("You're old enough to drive legally.")
+
+#step4: set up second conditional if user enters a number less than 0
+if user_age < 0: 
+    print("Please enter a valid age.")
+
+#step5: set up conditional if user is older than 0 but less than 16
```
Comment (paulghaddad) [inline ex16_ch4.py]: If you move the check for the invalid age as the first check, then you can do an `if`-`else` instead:

```
if user_age >= LEGAL_DRIVING_AGE:
  print("You're old enough to drive legally.")
else:
  print("You're not old enough to drive legally.")
```

This is more idiomatic as it's essentially a fork in the branch, which a `if`-`else` represents well.

**Comment by amymhaddad** [comment]
@paulghaddad I updated this exercise.

File: ex16_ch4.py (hunk 3 above)
Comment (amymhaddad) [reply ex16_ch4.py]: @paulghaddad I made this an if-else in my update. However, if the user enters an invalid number (ie, -1) then the "else" statement is still executed. Why is this?

**Pull request closed without merging**